### PR TITLE
Update 3.3.0

### PR DESCRIPTION
**Pull request by Cyteon**
# Description

Merge develop to main

# Issue Fixes

None

# Checklist:

- [ ] I am pushing changes to the **develop** branch
- [X] I am using the recommended development environment
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have formatted and linted my code using python-black and pylint
- [X] I have cleaned up unnecessary files
- [X] My changes generate no new warnings
- [X] My changes follow the existing code-style
- [X] My changes are relevant to the project

# Any other information (e.g how to test the changes)

None

### CODE REVIEW

File: video_creation/final_video.py [hunk 1]
```diff
@@ -202,6 +262,26 @@ def make_final_video(
 
     image_clips = list()
 
+    Path(f"assets/temp/{reddit_id}/png").mkdir(parents=True, exist_ok=True)
+
+    # Copyright 2024 beingbored (aka. Tim), MIT License, permission granted to use, copy, modify, and distribute.
```
Comment (OpenSourceSimon) [inline video_creation/final_video.py:267]: Why's there a copyright comment here?

File: video_creation/screenshot_downloader.py [hunk 2]
```diff
@@ -75,7 +74,7 @@ def get_screenshots_of_reddit_posts(reddit_object: dict, screenshot_num: int):
         print_substep("Launching Headless Browser...")
 
         browser = p.chromium.launch(
-            headless=True
+            headless=False
```
Comment (OpenSourceSimon) [inline video_creation/screenshot_downloader.py:77]: Still need to test this, headless False will cause a lot of confusion for the user

**Review by OpenSourceSimon — COMMENTED** [review]
Looking good, still needs some testing. Definitely won't call this a V4, just continue with V3

**Comment by pippinmole** [comment]
Hi,

Currently trying to use `develop` branch as you're using the `new.reddit.com` URL. Looks like I'm getting an error here:

image

```json
{
 "allow_nsfw": false,
 "theme": "dark",
 "times_to_run": 15,
 "opacity": 0.9,
 "storymode": false,
 "storymodemethod": 1,
 "storymode_max_length": 1000,
 "resolution_w": "1080",
 "resolution_h": "1920",
 "zoom": 1.1,
 "channel_name": "",
 "background": {
  "background_video": "",
  "background_audio": "",
  "background_audio_volume": 0,
  "enable_extra_audio": false,
  "background_thumbnail": false,
  "background_thumbnail_font_family": "",
  "background_thumbnail_font_size": 96,
  "background_thumbnail_font_color": ""
 },
 "tts": {
  "voice_choice": "streamlabspolly",
  "random_voice": true,
  "elevenlabs_voice_name": "Adam",
  "elevenlabs_api_key": "REDACTED",
  "aws_polly_voice": "",
  "streamlabs_polly_voice": "",
  "tiktok_voice": "en_uk_003",
  "tiktok_sessionid": "REDACTED",
  "python_voice": "",
  "py_voice_num": "",
  "silence_duration": 0.05,
  "no_emojis": true
 }
}
```

**Comment by pippinmole** [comment]
Me again, looks like the issue stems from `getsize` being removed in newer versions of `Pillow`, as mentioned [here](https://github.com/tensorflow/models/issues/11040#issuecomment-1630502714). Quick fix would be to downgrade `Pillow`: `Pillow==9.5.0`, as this fixes things for me.

**Comment by JasonLovesDoggo** [comment]
> Me again, looks like the issue stems from `getsize` being removed in newer versions of `Pillow`, as mentioned [here](https://github.com/tensorflow/models/issues/11040#issuecomment-1630502714). Quick fix would be to downgrade `Pillow`: `Pillow==9.5.0`, as this fixes things for me.

Did the full fix!

**Pull request merged**